### PR TITLE
Fix/ad library updates

### PR DESCRIPTION
**Pull request by petermason**
#### Description

Fix nuk window object - malformed
Replace multiple useState objects with single Reducer

#### Screenshots 

N/A

#### Checklist
 - [x] Unit tests
 - [ ] Updated E2E tests
 - [ ] Updated storybook
 - [ ] Tested on iOS simulator
 - [ ] Tested on Android emulator
 - [ ] Tested on Tablet

### CODE REVIEW

File: packages/ad/src/dom-context.tsx [hunk 1]
```diff
@@ -169,16 +188,16 @@ const DOMContext = (props: DomContextType) => {
...
         "ads": {
           "blocked": false,
-          "commercialSection": ${sectionName},
+          "commercialSection": "${sectionName}",
           "tuples": {
             "cont": "${slotName === "ad-section" ? "sec" : "art"}",
             "path": "",
             "cpn": "${String(authId)}",
-            "device": ${isTablet() ? "tablet" : "mobile"},
+            "device": "${isTablet() ? "tablet" : "mobile"}",
```
Comment (josholdham) [inline packages/ad/src/dom-context.tsx:200]: this looks odd - like it'd give double quotation marks - so just want to check it's needed

File: packages/ad/src/reducer.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      return {
+        ...state,
+        padding: action.payload,
+      };
+    default:
+      throw new Error("Unknown action");
+  }
+};
+
+export default reducer;
```
Comment (josholdham) [inline packages/ad/src/reducer.ts:59]: nice

**Review by josholdham — APPROVED** [review]
nice use of useReducer, haven't used it myself so good to learn from this!

File: packages/ad/src/dom-context.tsx (hunk 1 above)
Comment (petermason) [reply packages/ad/src/dom-context.tsx:200]: Good shout. Going to check with the ad team. Doesn't make sense to be empty

File: packages/ad/src/reducer.ts (hunk 2 above)
Comment (petermason) [reply packages/ad/src/reducer.ts:59]: I think useReducer is a lot cleaner and easier to understand when more that a couple of useState appear in the code. glad you like it

File: packages/ad/src/dom-context.tsx [hunk 3]
```diff
@@ -108,18 +118,27 @@ const DOMContext = (props: DomContextType) => {
   const handleMessageEvent = (event: WebViewMessageEvent) => {
     try {
       const data = JSON.parse(event.nativeEvent.data);
-      //console.log("AD DATA: ", data);
       switch (data.type) {
-        case "slotOnLoad":
-          setAdHeight(data.size);
-          setPadding(20);
+        case "slotOnLoad": {
+          console.log("slot on load", data.size);
```
[the remark this answers is not in the document]
Comment (petermason) [reply packages/ad/src/dom-context.tsx]: removed

File: packages/ad/src/dom-context.tsx (hunk 1 above)
Comment (petermason) [reply packages/ad/src/dom-context.tsx:201]: this needed to be the 'sectionname/slug-url' e.g 'news/liz-truss-keir-starmer-pmqs-bank-of-england-pensions-kwarteng-mini-budget-latest-news'

Updated to pass this value in

**Pull request merged**